extensions/common/gcp/gcp-core/src/main/java/org/eclipse/edc/gcp/iam/IamServiceImpl.java

```diff
@@ -196,4 +243,24 @@ private Supplier<IamCredentialsClient> defaultIamCredentialsClientSupplier() {
             };
         }
     }
+
+    static class ApplicationDefaultCredentials {
```

Comment:
seems kinda spurious, it's either only for internal use, in which case it should be `private`, or for public use, in which case I'd give it a dedicated class.

Reply:
The objective was to mock GoogleCredentials and test with application default credentials (e.g. valid ones, error, etc), I will search for a better way to test (pls let me know if there is a clear way to do that)